consumer.go

```diff
@@ -93,21 +92,50 @@ func NewConsumerForRemote(ctx context.Context, remote *bfs.Object, opt *Consumer
 		cfn:    cfn,
 	}
 
-	// run initial sync
-	if _, err := c.sync(true); err != nil {
-		_ = c.Close()
+	return c.run()
+}
+
+// NewIncrementalConsumer starts a new incremental feed consumer.
+func NewIncrementalConsumer(ctx context.Context, bucketURL string, opt *ConsumerOptions, cfn ConsumeFunc) (Consumer, error) {
+	bucket, err := bfs.Connect(ctx, bucketURL)
+	if err != nil {
 		return nil, err
 	}
 
-	// start continuous loop
-	go c.loop()
+	csm, err := NewIncrementalConsumerForBucket(ctx, bucket, opt, cfn)
+	if err != nil {
+		_ = bucket.Close()
+		return nil, err
+	}
+	csm.(*consumer).ownRemote = true
```

Comment:
something doesn't add up here, this should be `ownBucket` as we create a bucket here that needs to be closed